mm/kfence/core.c

```diff
@@ -534,9 +531,8 @@ bool kfence_handle_page_fault(unsigned long addr)
 			}
 		}
 		if (report_index != -1) {
-			object = kfence_metadata[report_index];
+			kfence_report_error(addr, &kfence_metadata[report_index], KFENCE_ERROR_OOB);
```

Comment:
Can we guarantee that kfence_report_error() won't allocate memory, causing a deadlock on kfence_alloc_lock?
This is why we've been passing the copy of kfence_metadata[report_index] to kfence_report_error().

Reply:
Yes I think so. printk should never allocate memory (otherwise a lot more places in the kernel might be broken), and we don't use anything else in there that might.